lib/urlwatch/reporters.py

```diff
@@ -978,3 +978,65 @@ def submit(self):
 
         for chunk in chunkstring(text, self.MAX_LENGTH, numbering=True):
             asyncio.run(xmpp.send(chunk))
+
+class ProwlReport(TextReporter):
+    """Send a detailed notification via prowlapp.com"""
+
+    __kind__ = 'prowl'
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def submit(self):
+        api_base = 'https://api.prowlapp.com/publicapi/'
+        api = {
+            'add': api_base + 'add'
+        }
+
+        text = '\n'.join(super().submit())
+
+        if not text:
+            logger.debug('Not calling Prowl API (no changes)')
+            return
+
+        filtered_job_states = list(self.report.get_filtered_job_states(self.job_states))
+        subject_args = {
+            'count': len(filtered_job_states),
+            'jobs': ', '.join(job_state.job.pretty_name() for job_state in filtered_job_states),
+        }
+
+        # 'subject' used in the config file, but the API
+        # uses what might be called the subject as the 'event'
+        event = self.config['subject'].format(**subject_args)
+
+        # 'application' is prepended to the message in prowl,
+        # to show the source of the notification. this too,
+        # is user configurable, and may reference subject args
+        application = self.config['application'].format(**subject_args)
+
+        # if the user didn't set application, just use the package name and version.
+        if not application:
+            application = ('{0} v{1}'.format(urlwatch.pkgname, urlwatch.__version__))[:256].encode('utf8')
+
+        # build the data to post
+        post_data = {
+            'event': event[:1024].encode('utf8'),
+            'description': text[:10000].encode('utf8'),
+            'application': application[:256].encode('utf8'),
```

Comment:
I'm pretty sure that you don't want to encode the data as UTF-8 here, as the requests posting code should take care of that, right?